examples/step-57/doc/intro.dox

```diff
@@ -0,0 +1,336 @@
+- \nabla \cdot \textbf{u} &=& 0.
+@f}
+
+acts as the initial guess for
+
+@f{eqnarray*}
+- \nu_{i+1} \Delta\textbf{u} + (\textbf{u} \cdot \nabla)\textbf{u} + \nabla p &=& \textbf{f}\\
+- \nabla \cdot \textbf{u} &=& 0
+@f}
+with $|\nu_{i+1} - \nu_{i}|$ small enough. 
+
+We repeat this process till getting the solution to the NSE with viscosity
+$\nu_{n-1}$, then the initial guess for the target NSE is obtained.
+
+<h3>The Solver and Preconditioner </h3>
+
+At each step of Newton's iteration, the problem results in solving a
+saddle point systems of the form
+@f{eqnarray*}
+    \left(
+      \begin{array}{cc}
+        A & B^{T} \\
+        B & 0 \\
+      \end{array}
+    \right)
+    \left(
+      \begin{array}{c}
+        U \\
+        P \\
+      \end{array}
+    \right)
+    =
+    \left(
+      \begin{array}{c}
+        F \\
+        0 \\
+      \end{array}
+    \right).
+@f}
+
+Instead of solving the above system, we can solve the equivalent system
+
+@f{eqnarray*}
+    \left(
+      \begin{array}{cc}
+        A + \gamma B^TW^{-1}B & B^{T} \\
+        B & 0 \\
+      \end{array}
+    \right)
+    \left(
+      \begin{array}{c}
+        U \\
+        P \\
+      \end{array}
+    \right)
+    =
+    \left(
+      \begin{array}{c}
+        F \\
+        0 \\
+      \end{array}
+    \right)
+@f}
+with a parameter $\gamma$ and an invertible matrix W.
+
+Denoting the system matrix of the new system by $G$ and right hand
+side by $b$, we solve it iteratively with right preconditioning
+$P^{-1}$ as $GP^{-1}y = b$, where
+
+@f{eqnarray*}
+P^{-1} = \left(\begin{array}{cc} \tilde{A} & B^T \\ 
+                                 0 & \tilde{S} \end{array}\right)^{-1},
+@f}
+
+with $\tilde{A} = A + \gamma B^TW^{-1}B$ and $\tilde{S}$ is the
+corresponding Schur complement $\tilde{S} = B^T \tilde{A}^{-1} B$. we
+let $W = M_p$ where $M_p$ is the pressure mass matrix, then
+$\tilde{S}^{-1}$ can be approximated by
+
+@f{eqnarray*}
+\tilde{S}^{-1} \approx -(\nu+\gamma)M_p^{-1}.
+@f}
+See "An Augmented Lagrangian-Based Approach to the Ossen Problem,
+M.Benzi and M.Olshanskii" for details.
+
+We decompose $P^{-1}$ as
+
+@f{eqnarray*} 
+P^{-1} =
+\left(\begin{array}{cc} \tilde{A}^{-1} & 0 \\ 0 & I \end{array}\right)
+\left(\begin{array}{cc} I & -B^T \\ 0 & I \end{array}\right)
+\left(\begin{array}{cc} I & 0 \\ 0 & \tilde{S}^{-1} \end{array}\right).
+@f}
+
+Here two inexact solvers will be needed for $\tilde{A}^{-1}$ and
+$\tilde{S}^{-1}$. See [1]. Since the pressure mass
+matrix is symmetric, the solver for $\tilde{S}^{-1}$ is not diffcult
+and we can use CG with ILU as preconditionerr. We use the direct
+solver UMFPACK for $\tilde{A}^{-1}$. The remaining is a sparse
+mastrix-vector product with $B^T$. Further more, the augmented
+lagrangian term will be assembled through Grad-Div stabilization:
+$(\nabla \cdot \phi _{j}, \nabla \cdot \phi _{i}) \approx (B^T
+M_p^{-1}B)_{ij}$. See [2].

```

Comment:
I would suggest that you also compare to UMFPACK on the whole matrix. Since I'm rather a practitioner than a developer for Schur complement approximations, I must admit I'm getting nervous about Schur complement approaches that use a direct solver inside (for the big block). If you factorize 90% of the matrix, why not go for the remaining 10% as well that increase cost by say 2% but avoid doing 5 iterations and the whole GMRES stuff? Alternatively, I would be curious if you have a good approach for your A^-1 which is lightweight (i.e., at most a multigrid application) ;)